Add Homepage UI

priyanshisharma opened the pull request:
### Fixes #65 

**Description** 
I created the Homepage as per the Figma Design.


**Screenshots** 
<img width="1440" alt="Screen Shot 2020-09-20 at 8 21 25 PM" src="https://user-images.githubusercontent.com/30956619/93714238-f3071480-fb7e-11ea-9f10-8863820ec7e6.png">
<img width="1440" alt="Screen Shot 2020-09-20 at 8 21 34 PM" src="https://user-images.githubusercontent.com/30956619/93714246-087c3e80-fb7f-11ea-87f3-f4b4c631d65c.png">
<img width="1440" alt="Screen Shot 2020-09-20 at 8 21 41 PM" src="https://user-images.githubusercontent.com/30956619/93714247-09ad6b80-fb7f-11ea-851f-3e4099028edf.png">
<img width="1440" alt="Screen Shot 2020-09-20 at 8 21 48 PM" src="https://user-images.githubusercontent.com/30956619/93714248-0a460200-fb7f-11ea-93d4-4e127bca6374.png">

salil-naik commented on frontend/src/css/components/blogCard.scss:
```diff
@@ -3,6 +3,7 @@
 .blog-card {
   &__container {
     border-radius: 5px;
+    width: 340px;
```
Do not give it a fixed with. Blog card will take the width of the parent element. 

```
<div className="cell large-4>
    <BlogCard />
</div>
```

priyanshisharma replied:
Okay, got it 🙌

priyanshisharma replied:
Okay, got it 👍

salil-naik commented on frontend/src/css/pages/home.scss:
```diff
@@ -0,0 +1,58 @@
+@import '../components/_button';
+
+.intro{
+    height: 616px;
```
Never give fixed height and width. Height will increase or decrease automatically depending on the content. Width will increase/decrease depending on the screen size.

priyanshisharma replied:
Okay.. shall I use padding instead?

salil-naik commented on frontend/src/css/pages/home.scss:
```diff
@@ -0,0 +1,58 @@
+    align-items: center;
+}
+.importanceOfDiversity{
+    width: 1280px;
```
height & width

salil-naik commented on frontend/src/css/pages/home.scss:
```diff
@@ -0,0 +1,58 @@
+    
+}
+.feature{
+    width: 1280px;
```
height & width

salil-naik commented on frontend/src/css/pages/home.scss:
```diff
@@ -0,0 +1,58 @@
+}
+
+
+.list{
```
list class should always be used for lists. Do not use it.

priyanshisharma replied:
Okay, got it 👍

priyanshisharma replied:
Removed 👍

salil-naik commented on frontend/src/pages/home.jsx:
```diff
@@ -0,0 +1,162 @@
+            </div>
+            <div style={{background:"linear-gradient(180deg, #FFFDF6 0%, #FFEBF3 100%)"}}>
+                <div className="importanceOfDiversity ">
+                    <div className="cell small-7 large-offset-1">
```
offsets are not required anywhere for the homepage. Kindly remove all the offsets.

priyanshisharma replied:
Okay, got it 👍

priyanshisharma replied:
Done 👍

salil-naik commented on frontend/src/pages/home.jsx:
```diff
@@ -0,0 +1,162 @@
+                            Read the impact stories from around the world
+                        </h6>
+                        <br/>
+                        <ul className="list__item-container large-offset-1">
```
This markup will create a problem while making the site responsive. 

It should be like this
```
<div className="grid-container">
    <div className="grid-x grid-margin-x">
        <div className="cell large-12">
            //heading and text will come here
        </div>

         <div className="cell large-4">
            <BlogCard />
        </div>
        <div className="cell large-4">
            <BlogCard />
        </div> 
        <div className="cell large-4">
            <BlogCard />
        </div>
    </div>
</div>

priyanshisharma replied:
Got it, thank you!

priyanshisharma replied:
Okat, understood 👍

salil-naik commented on frontend/src/pages/home.jsx:
```diff
@@ -0,0 +1,162 @@
+                            <br/>
+                            <ul className="list__item-container large-offset-1 centreitems">
+                                <li className="list__item">
+                                    <img src={GoogleAd} width="254.41px" height="80px"  alt="Google Ads"/>
```
Never give fixed widths and heights. Everything should be relative (some exceptions are there, like logo). If you give fixed dimensions, it will break on smaller screens.

priyanshisharma replied:
I understand that the width is taken care of the parent element, eg. by using `cell large-4`, but what about height?

priyanshisharma replied:
Okay, got it 👍

salil-naik reviewed (changes requested):
Hi. A lot of things have to be changed. And use BEM strictly. 

For example, 

```
.hero{
    &__title{
        //this is for the heading (h1)
    }
    
    &__text{
    }

    &__button-contaier{
    }
}
````

priyanshisharma commented:
* For BEM, I have not made classes for individual elements in the `div` tags, so what am I supposed to divide? Or is it some naming convention.
* I have removed `width` and `height` from everywhere. 
* Without the height, `div`s left no space in between, and so **I've added `padding` for now. Is that alright?**
* I have removed the `list` class in CSS.
* I've removed the offsets.
* **How to manage height of image elements?** This is how that section looks like right now.
<img width="1355" alt="Screen Shot 2020-09-24 at 8 36 07 PM" src="https://user-images.githubusercontent.com/30956619/94163569-a33d8b80-fea5-11ea-9a60-aad66343201d.png">

salil-naik commented:
BEM is more of a naming convention. 
Padding should be set **only** on the top and bottom of the sections (divs). The height should always be auto. If we decide to add more content, the auto height will make sure that it takes the height as per the content and padding on top and bottom will leave enough breathing space. 

For the partner logos, set the max-width of the logos to 250px; Give `display: flex` to the parent (logo container) and justify the content to center.

priyanshisharma commented:
I think I got it, please have a look at the code.

This is how the `Our Partners` section look like right now.
<img width="1440" alt="Screen Shot 2020-09-25 at 12 35 19 AM" src="https://user-images.githubusercontent.com/30956619/94188529-26230e00-fec7-11ea-9209-71f82d5955b5.png">
